spyder_kernels/customize/code_runner.py

```diff
@@ -0,0 +1,593 @@
+            return
+
+        if not cell_code or cell_code.strip() == "":
+            print("Nothing to execute, this cell is empty.\n")
+            return
+
+        # Trigger `post_execute` to exit the additional pre-execution.
+        # See Spyder PR #7310.
+        self.shell.events.trigger("post_execute")
+        file_code = self._get_file_code(filename, save_all=False)
+
+        # Here the remote filename has been used. It must now be valid locally.
+        filename = canonic_filename
+
+        with NamespaceManager(
+            self.shell,
+            filename,
+            current_namespace=True,
+            file_code=file_code,
+            context_locals=context_locals,
+            context_globals=context_globals
+        ) as (ns_globals, ns_locals):
+            return self._exec_code(
+                cell_code,
+                filename,
+                ns_globals,
+                ns_locals,
+                post_mortem=post_mortem,
+                exec_fun=exec_fun,
+                capture_last_expression=True,
+            )
+
+    def _get_current_file_name(self):
+        """Get the current editor file name."""
+        try:
+            return frontend_request(blocking=True).current_filename()
+        except Exception:
+            print(
+                "This command failed to be executed because an error occurred"
+                " while trying to get the current file name from Spyder's"
+                " editor. The error was:\n\n"
+            )
+            self.shell.showtraceback(exception_only=True)
+            return None
+
+    def _get_file_code(self, filename, save_all=True, raise_exception=False):
+        """Retrieve the content of a file."""
+        # Get code from spyder
+        try:
+            return frontend_request(blocking=True).get_file_code(
+                filename, save_all=save_all
+            )
+        except Exception:
+            # Maybe this is a local file
+            try:
+                with open(filename, "r") as f:
+                    return f.read()
+            except FileNotFoundError:
+                pass
+            if raise_exception:
+                raise
+            # Else return None
+            return None
+
+    def _exec_code(
+        self,
+        code,
+        filename,
+        ns_globals,
+        ns_locals=None,
+        post_mortem=False,
+        exec_fun=None,
+        capture_last_expression=False,
+        global_warning=False,
+    ):
+        """Execute code and display any exception."""
+        global SHOW_INVALID_SYNTAX_MSG
+        global SHOW_GLOBAL_MSG
+
+        if exec_fun is None:
+            exec_fun = exec
+
+        is_ipython = os.path.splitext(filename)[1] == ".ipy"
+        try:
+            if not is_ipython:
+                # TODO: remove the try-except and let the SyntaxError raise
+                # Because there should not be ipython code in a python file
```

Comment:
```suggestion
                # TODO: Remove the try-except and let the SyntaxError raise
                # because there should't be IPython code in a Python file.
```